### PR TITLE
Unit Test: Bolt_Boltpay_Model_Admin_ExtraConfig

### PR DESCRIPTION
**Pull request by mccottry**
# Description
Unit Test: Bolt_Boltpay_Model_Admin_ExtraConfig

Fixes: https://app.asana.com/0/941895179897714/1155250167569193

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+    /**
+     * @test
+     * that hasValidBoltPrimaryColor returns false for invalid color hex and adds error message to session
+     *
+     * @covers ::hasValidBoltPrimaryColor
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidBoltPrimaryColor_withInvalidColorHex_returnsFalseAndAddsErrorMessageToSession()
```
Comment (johnmccombs1) [inline tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: @mccottry Can you add a new unit test named hasValidBoltPrimaryColor_withValidColorHex_returnsTrue?

File: tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+    /**
+     * @test
+     * that hasValidDatadogKeySeverity returns false for invalid severity keys and adds error message to session
+     *
+     * @covers ::hasValidDatadogKeySeverity
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidDatadogKeySeverity_withInvalidSeverityKey_returnsFalseAndAddsErrorMessageToSession()
```
Comment (johnmccombs1) [inline tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: @mccottry Can you add a new unit test named hasValidDatadogKeySeverity_withValidSeverityKey_returnsTrue?

File: tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+    /**
+     * @test
+     * that hasValidPriceFaultTolerance returns false for negative integers and adds error message to session
+     *
+     * @covers ::hasValidPriceFaultTolerance
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidPriceFaultTolerance_withNegativeInteger_returnsFalseAndAddsErrorMessageToSession()
```
Comment (johnmccombs1) [inline tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: @mccottry Can you add a new unit test named hasValidPriceFaultTolerance_withValidValue_returnsTrue?

Comment (mccottry) [reply tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: I can update the test for you.

File: tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php (hunk 2 above)
Comment (mccottry) [reply tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: I can update the test for you.

File: tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php (hunk 1 above)
Comment (mccottry) [reply tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php]: I can update the test for you.

**Comment by mccottry** [comment]
@johnmccombs1 I made the test that you are looking for more clear and more thorough in the dataProvider method `save_withValidParameters_savesNewValueProvider`.

Let me know if this resolves your concerns.

**Comment by johnmccombs1** [comment]
LGTM

**Pull request merged**